### PR TITLE
Move UICollectionView(DataSource|Delegate) to category

### PR DESCRIPTION
**Pull request by amonshiz**
This reduces the quantity of the code within IGListAdapter.m and is a bit more in line with Swift style of putting protocol conformance into extensions.

## Changes in this pull request
This does not change functionality just rearranges where code lives.

### Checklist

- [X] All tests pass. Demo project builds and runs.
- [X] I added tests, an experiment, or detailed why my change isn't tested.
- [N/A] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [N/A] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: Examples/Examples-iOS/Podfile.lock [hunk 1]
```diff
@@ -17,4 +17,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 55363d0329e656c259d9b9982b162afd8d8904c6
 
-COCOAPODS: 1.2.0
+COCOAPODS: 1.2.0.rc.1
```
Comment (jessesquires) [inline Examples/Examples-iOS/Podfile.lock]: let's do this with `1.2.0`

Comment (amonshiz) [inline Examples/Examples-iOS/Podfile.lock]: okay i have no idea how i got that. XD

**Comment by rnystrom** [comment]
We should add a `#refactor` or something tag so Danger chills out.

@amonshiz mind rebasing and fixing the example project files? You can just merge w/e and run the `pod_setup.sh` script again to overwrite them.

**Comment by jessesquires** [comment]
lol this is because I landed #409 so now everyone has conflicts 😄 

you're welcome

**Comment by amonshiz** [comment]
Gonna wait for #608 to land and then rebase on top of that.

**Comment by jessesquires** [comment]
@amonshiz - is this good to go? need to rebase?

**Comment by jessesquires** [comment]
Oh -- it would be nice to right click > "sort by name" on all files in the project 😄

**Comment by amonshiz** [comment]
definitely need to rebase.

i'll try to get the files ordered properly. should i ensure that all folders are at the top?

**Comment by jessesquires** [comment]
@amonshiz - want to try to land this today? 😊

**Comment by amonshiz** [comment]
just rebasing

**Pull request closed without merging**